ClinGen/clincoded
# Connects to #1116 kl update casecontrol

This PR contains UI updates in case control page, including:
 1. Used message "Number only" to replace "e.g. number only".
 2. Changed type of "Other test statistic" to text.
 3. Set a method to check fields that only take number and to display different messages to integer and decimal values.
 4. Removed pull-down list "Type" under `Age Range` from control cohort.
 5. Set status to "deleted" to case cohort and control cohort when their parent object case control is deleted

**Test 1 -- Message and data field updates**
* Create a GDM, add a PMID, add a Case-Control and **expect to see:**
  1. There is no "e.g. number only" in any input box,
  2. There is no field `Type` under `Age Range` in Control Cohort.
![screen shot 2016-11-02 at 11 38 31 am](https://cloud.githubusercontent.com/assets/9206696/19942429/eed6affc-a0f0-11e6-9698-aca78b565955.png)

* Enter "casecontrol AABB" to `Case-Control Label`, "case AA" to `Case Cohort Label`, "control BB" to `Control Cohort Label` and "ORPHA15" to `Disease(s) in Common ` in Case Cohort;

* In both Case Cohort and Control Cohort, enter a decimal number, for example 12.5, in fields `Number of males` and `Number of females` under Demographics, and in fields `Number of Cases with variant(s) in the gene in question` and `Number of all Cases genotyped/sequenced` under Power;

* Enter "abc" in field `Case Allele Frequency` and "xyz" in field `Control Allele Frequency`.

* In Case-Control Evaluation & Score, select "Others" at `Test statistic` and then enter a string, for example "some thing here", in field `Other test statistic`;

* Enter "abc" in field `Value`, "xyz" in field `p-value`, "a" in field `Confidence interval (%)` and "z" in field left.

* Click Save and **expect to see:**
  1. Error messages "Non-decimal values only" under fields with decimal numbers, "Number only" under fields with non-digit text.
![screen shot 2016-11-02 at 11 23 39 am](https://cloud.githubusercontent.com/assets/9206696/19941832/d628942c-a0ee-11e6-9fd8-0b7f20464959.png)
![screen shot 2016-11-02 at 8 18 34 pm](https://cloud.githubusercontent.com/assets/9206696/19954860/90c5981c-a139-11e6-918a-4f9088a54d9c.png)
![screen shot 2016-11-02 at 8 20 01 pm](https://cloud.githubusercontent.com/assets/9206696/19954874/c3edc43a-a139-11e6-8283-97e609740b82.png)

  2. There is no error message for field `Other test statistic`.
![screen shot 2016-11-02 at 11 32 07 am](https://cloud.githubusercontent.com/assets/9206696/19942199/2d111466-a0f0-11e6-89b0-a793423df972.png)

* Change decimal numbers to integers in all fields with error message "Non-decimal values only" and change text to a number in all fields with message "Number only", click Save again and **expect to see `casecontrol AABB` saved properly**.


**Test 2 -- Delete Case-Control**
* In curation-central page for the GDM, click Edit under `casecontrol AABB` as:
![screen shot 2016-11-02 at 11 44 21 am](https://cloud.githubusercontent.com/assets/9206696/19942701/e3c10ae4-a0f1-11e6-90ca-bc6f478392a2.png)

* Click Delete and **expect to see**
![screen shot 2016-11-02 at 11 48 44 am](https://cloud.githubusercontent.com/assets/9206696/19942847/5c359832-a0f2-11e6-91df-8503a4e988d0.png)

* Click Confirm Delete and **expect to go back to curation-central page**

* Go to [http://localhost:6543/casecontrol/](http://localhost:6543/casecontrol/), click "casecontrol AABB", add "?format=json" at the end of url, return and **expect to see `status: "deleted",` in case control, case cohort and control cohort**

**End of Teset**
